lib.go

```diff
@@ -940,6 +941,28 @@ func setPtrFieldsVal(fields []string, immutable, mutable reflect.Value) {
 	}
 }
 
+// utf16UnitCountInString returns the number of UTF-16 code units in a string.
```

Comment:
This function can be simplify by `len(utf16.Encode([]rune(s)))`.

Reply:
Thank you for the review and merge!

I think this is fine if the priority is to keep the code simple.

Just to add a note: since both []rune and []uint16 are allocated in memory at the same time, the original code was written to minimize memory usage in case a long string is passed in.

Reply:
Thanks for your notice, I missed that point, will back port for it.